pkg/xds/envoy/listeners/filter_chain_configurers.go

```diff
@@ -86,11 +85,18 @@ func ServerSideStaticTLS(tlsCerts core_xds.ServerSideTLSCertPaths) FilterChainBu
 	})
 }
 
-func HttpConnectionManager(statsName string, forwardClientCertDetails bool) FilterChainBuilderOpt {
-	return AddFilterChainConfigurer(&v3.HttpConnectionManagerConfigurer{
+func HttpConnectionManager(statsName string, forwardClientCertDetails bool, annotators ...v3.HttpConnectionManagerConfigurerAnnotateFunc) FilterChainBuilderOpt {
+	hcmConfigurer := &v3.HttpConnectionManagerConfigurer{
```

Comment:
Adding optional "annotator" functions to this Opt so that existing invocation can stay unchanged. Existing code on mutating the HttpConnectionManager by marshal+unmarshal the proto message could potentially benifit from this design too.

Reply:
It's refactored, so no more annotators